### PR TITLE
fix hanging mouse in piano roll (#4822)

### PR DESCRIPTION
**Pull request by sharpblade4**
Hey, this is my first issue. I'll appreciate any comment. Hope I did it fine.

Edit: Fixes https://github.com/LMMS/lmms/issues/4822

### CODE REVIEW

**Comment by SecondFlight** [comment]
Style looks fine to me 👍

**Comment by SecondFlight** [comment]
@sharpblade4 Does this fix a specific issue that you could link?

**Comment by qnebra** [comment]
This I think:

https://github.com/LMMS/lmms/issues/4822

**Comment by sharpblade4** [comment]
yes, #4822 . thanks for the edit.

**Comment by BaraMGB** [comment]
Works for the upper side of piano roll, but not for the left side.

File: src/gui/editors/PianoRoll.cpp [hunk 1]
```diff
@@ -1294,6 +1294,7 @@ void PianoRoll::leaveEvent(QEvent * e )
 
 	QWidget::leaveEvent( e );
 	s_textFloat->hide();
+	this->update(); // cleaning inner mouse-related graphics
```
Comment (BaraMGB) [inline src/gui/editors/PianoRoll.cpp]: No need for `this->`.

**Comment by sharpblade4** [comment]
I'll work on resolving this, as well as fixing other windows (automation & mixer) that were recently mentioned in the bug report.  When I'm finished, should I post a new pull request? or somehow add it to this one?

**Comment by zonkmachine** [comment]
> When I'm finished, should I post a new pull request? or somehow add it to this one?

You commit these fixes to the same branch and push it to `sharpblade4:stable-1.2`.
This will also update the PR.

**Comment by sharpblade4** [comment]
PR updated with the required improvements. Couldn't reproduce the bug in the mixer. I'll ask for more details in the original issue thread.

**Comment by zonkmachine** [comment]
> PR updated with the required improvements

Yes, it now works also for the left side and in the Automation Editor.
Thanks for working on this, fix confirmed!

File: src/gui/editors/AutomationEditor.cpp [hunk 2]
```diff
@@ -1510,8 +1510,12 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
 		case SELECT: cursor = s_toolSelect; break;
 		case MOVE: cursor = s_toolMove; break;
 	}
-	p.drawPixmap( mapFromGlobal( QCursor::pos() ) + QPoint( 8, 8 ),
-								*cursor );
+	// TODO RON copy all if, replace the top left etc
```
Comment (BaraMGB) [inline src/gui/editors/AutomationEditor.cpp]: Did you forget to remove this?

Comment (sharpblade4) [reply src/gui/editors/AutomationEditor.cpp]: so sorry, totally forgot that line there.

File: src/gui/editors/PianoRoll.cpp (hunk 1 above)
Comment (sharpblade4) [reply src/gui/editors/PianoRoll.cpp]: fixed

**Comment by sharpblade4** [comment]
I deleted a comment line that I left in the code by mistake, I did it through the github website rather than pushing this change from the computer. Is that the problem that caused continuous-integration/travis-ci/pr to fail?

**Comment by zonkmachine** [comment]
> Is that the problem that caused continuous-integration/travis-ci/pr to fail?

I restarted the build and now it passed. I didn't look into what made the build fail but it's usually something like a timeout or it can't download a dependency.

**Comment by zonkmachine** [comment]
Merge?

**Comment by BaraMGB** [comment]
Looks good to me.

**Pull request merged**